packages/backend/test/toolExecutionParity.test.ts

```
@@ -106,109 +47,258 @@ test('weather success keeps tool status executed and still runs generation', asy
+    );
+    const toolStep = result.workflowLineage.steps.find(
+        (step) => step.stepKind === 'tool'
+    );
+    assert.ok(toolStep, 'Workflow should have a tool step');
+    assert.equal(
+        toolStep?.outcome.status,
+        'executed',
+        'Tool step should be executed'
+    );
+    assert.ok(
+        toolStep?.outcome.artifacts?.length ?? 0 > 0,
```

Biome (2.4.13)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)